app/helpers/helper.rb

```diff
@@ -196,5 +196,23 @@ def self.save_transcript(transcript)
       secret
     end
 
+    def self.formatted_display_name(user)
+      if !SiteSetting.enable_names || user.name.blank?
+        return "@#{user.username}"
+      end
+
+      full_name = user.name
+
+      similar = (full_name.strip.gsub(' ', '_').casecmp(user.username) == 0) || (full_name.strip.gsub(' ', '').casecmp(user.username) == 0)
```

Comment:
Should this work with Unicode usernames? `casecmp` supports only A-Z.

```suggestion
      full_name_normalized = User.normalize_username(full_name.strip)
      similar = full_name_normalized.gsub(' ', '_') == user.username_lower  || full_name_normalized.gsub(' ', '') == user.username_lower
```